pkg/frontend/transport/handler.go

```diff
@@ -88,7 +88,7 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if f.cfg.QueryStatsEnabled {
 		var ctx context.Context
 		stats, ctx = querier_stats.ContextWithEmptyStats(r.Context())
-		r = r.WithContext(ctx)
+		*r = *r.WithContext(ctx)
```

Comment:
This seems very brittle... it may easily break if we insert another middleware between this handler, and ServerTimingMiddleware.

We could move generation of server timing header into this handler.

Alternatively we could move stats logging into the same middleware as server-timing header.